### ISSUE TITLE
Add more Sassdoc annotations and generate Sass docs in markdown

### ISSUE THREAD

**Issue by mattrosno** [carbon-design-system/carbon#2266]
### Summary

We're currently using Sassdoc to mark public or private `@access` but beyond that, we're inconsistent and documenting Sass hasn't been a priority.

Once https://github.com/IBM/carbon-elements/pull/462 is merged and subsequently merged into `carbon-components`, we can modify our build to auto-generate our public Sass API.

Here's an example from `carbon-elements` once all Sass functions, mixins, and variables have been annotated: https://github.com/IBM/carbon-elements/blob/0f6ffed16a7aa0b3074a665ea7b54184a5bec7b6/packages/elements/docs/sass.md

There would be three parts to completing this:

1. Add complete, consistent, Sassdoc annotations to all Sass source originating from `carbon-components`
1. Document Sassdoc annotation style in our contribution guidelines
1. New build script to call in our built, e.g.:

```json
"scripts": {
  "build": "gulp build && yarn docs",
  "docs": "bundler sassdoc 'scss' --output 'docs/api' && bundler sassdoc 'scss' --json --output 'docs/api/data'"
}
```

That example above generates `sass.md` in the `/docs/api` directory, and also `sass.json` in `/docs/api/data` directory. `sass.json` is the data dump generated by Sassdoc during the build.

**Not in scope for this issue**: Going forward, it would be great to have `carbon-website` consume the json object and render a more visually appealing way to understand and use our tokens.

### Justification

Our Sass is minimally documented and even with the current comments, you have to crawl through source to find it.

By having one source of truth (`/docs/api/sass.md`), a developer can more easily find what's available in our Sass API when using Carbon styling to extend Carbon through custom components or to theme Carbon.

**Not in scope for this issue**: Also, with `/docs/api/data/sass.json`, we'd be able to programmatically present token and theming information in `carbon-website`.

**Comment by asudoh** [comment]
@mattrosno The issue description refers to our website code - Was this meant to be for that repo? Thanks!

**Comment by mattrosno** [comment]
@asudoh I updated the issue description with two **not in scope for this issue**. This is not an issue for `carbon-website`. Just explaining how the outcome of completing this issue could benefit `carbon-website` in the future.

**Comment by asudoh** [comment]
@mattrosno Sounds that you are for generating SassDoc output from `carbon-components` and hosting it in `carbon-website`, is it correct? If so, FWIW `carbon-components` NPM package contains the raw Sass code with the SassDoc annotations, which means `carbon-website` can do everything by itself.

**Comment by mattrosno** [comment]
@asudoh regarding `carbon-website`, that works.

This issue focuses on `/docs/api/sass.md` that lives in *this* repo.

**Comment by asudoh** [comment]
OK cool - Thank you @mattrosno for clarifying!

**Comment by mattrosno** [comment]
@joshblack can you review the deprecated items in elements here? 

https://github.com/IBM/carbon-elements/blob/0f6ffed16a7aa0b3074a665ea7b54184a5bec7b6/packages/elements/docs/sass.md

Colors and themes packages had deprecated comments in v10 release so that 100% accurate. For layout, I designated the non-prefixed aliases as deprecated to hopefully encourage use of the prefixed variables? Can we do that? It makes the table of contents so much cleaner. Maybe in the deprecated comments we state when the deprecated variables will be available through? What version?
